docs/DeveloperGuide.md

```diff
@@ -1,34 +1,579 @@
+  - [Appendix: Requirements](#appendix-requirements)
+  - [Product scope](#product-scope)
+    - [Target user profile](#target-user-profile)
+      - [Value proposition](#value-proposition)
+    - [User Stories](#user-stories)
+    - [Non-Functional Requirements](#non-functional-requirements)
+    - [Glossary](#glossary)
+    - [Instructions for manual testing](#instructions-for-manual-testing)
+      - [Launch and shutdown](#launch-and-shutdown)
+      - [Adding a task](#adding-a-task)
+      - [Deleting a task](#deleting-a-task)
+      - [Marking a task as done](#marking-a-task-as-done)
+      - [Viewing the full details of a task](#viewing-the-full-details-of-a-task)
+      - [Editing a task](#editing-a-task)
+      - [Listing all tasks](#listing-all-tasks)
+      - [Help](#help)
+      - [Data storage](#data-storage-1)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Introduction
 
+Welcome to ATHENA's Developer Guide! ATHENA (which stands for Automated Timetable Helper Encourager n' Assistant), is a desktop daily life planner that aims to help students automate the process of organising their schedule. It is a Command Line Interface (CLI) based application that helps users figure out the best timetable after the user has input their pre-allocated time slots for work and relaxation.
+
+This document describes the software architecture and design of ATHENA that should hopefully help you - a developer, designer, or software tester - understand  the inner workings of ATHENA. 
+
+
+## Setting up and getting started
+
+### Prerequisites
+1. Ensure you have `JDK 11` installed on your computer.
+2. Install IntelliJ IDEA (referred to as IntelliJ in the rest of this document).
+
+### Setting up the project in your computer
+1. Fork [this](https://https://github.com/AY2021S1-CS2113T-W12-2/tp) repository, and clone the fork onto your computer.
+2. Open IntelliJ (If you do not see the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle  
+   1. Click `Configure` > `Project Defaults` > `Project Structure`.
+   2. Click `New…` and select the directory where you have installed `JDK 11`.
+   3. Click `OK`.
+4. By default, IntelliJ has the Gradle plugin installed. If you have disabled it, go to `File` > `Settings` > `Plugins` to re-enable it.
+5. Click `Import Project`(or `Open or Import` in newer versions of IntelliJ).
+6. Locate the `build.gradle` file and select it. Click `OK`. If prompted, choose to `Open as Project` (not `Open as File`).
+7. Click `OK` to accept the default settings, if prompted. 
+8. Wait for the importing process to finish, and you are good to go!
+9. Verify the setup:
+    1. Run `Athena` and try a few commands such as `list` or `help`.
+    2. Run the JUnit tests to ensure they all pass.
+
+## **Design & implementation**
+
+### Architecture
```

Comment:
Summary of Sequence Diagram: 
Should the naming of the header box be :Eample with underline?
Should the foot box be removed?
Some return line should come after the deactivation of the box?
Method names should be aligned with the arrow?
Example screenshots are provided below for reference